src/IECoreScene/ShaderNetworkAlgo.cpp

```diff
@@ -684,6 +713,213 @@ IECore::DataPtr loadSpline(
 
 } // namespace
 
+void ShaderNetworkAlgo::convertToOSLConventions( ShaderNetwork *network, int oslVersion )
+{
+	std::map< IECore::InternedString, std::pair< IECore::InternedString, size_t > > currentSplineArrayAdapters;
+
+	for( const auto &s : network->shaders() )
+	{
+		if( !boost::starts_with( s.second->getType(), "osl:" ) )
+		{
+			continue;
+		}
+
+		const auto &origParameters = s.second->parameters();
+		bool hasSplines = false;
+		for( const auto &i : origParameters )
+		{
+			if(
+				runTimeCast<const SplinefColor3fData>( i.second.get() ) ||
+				runTimeCast<const SplineffData>( i.second.get() )
+			)
+			{
+				hasSplines = true;
+				break;
+			}
+		}
+
+		if( !hasSplines )
+		{
+			continue;
+		}
+
+		CompoundDataPtr newParametersData = new CompoundData();
+		CompoundDataMap &newParameters = newParametersData->writable();
+
+		for( const auto &i : origParameters )
+		{
+			if( const SplinefColor3fData *colorSpline = runTimeCast<const SplinefColor3fData>( i.second.get() ) )
+			{
+				expandSpline( i.first, colorSpline->readable(), newParameters );
+			}
+			else if( const SplineffData *floatSpline = runTimeCast<const SplineffData>( i.second.get() ) )
+			{
+				expandSpline( i.first, floatSpline->readable(), newParameters );
+			}
+			else
+			{
+				newParameters[ i.first ] = i.second;
+			}
+		}
+
+		currentSplineArrayAdapters.clear();
+
+
+		std::vector< ShaderNetwork::Connection > connectionsToAdd;
+		ShaderNetwork::ConnectionRange inputConnections = network->inputConnections( s.first );
+		for( ShaderNetwork::ConnectionIterator it = inputConnections.begin(); it != inputConnections.end(); )
+		{
+			// Copy and increment now so we still have a valid iterator
+			// if we remove the connection.
+			const ShaderNetwork::Connection connection = *it++;
+
+			const std::string &destName = connection.destination.name.string();
+			boost::cmatch splineElementMatch;
+			if( !boost::regex_match( destName.c_str(), splineElementMatch, g_splineElementRegex) )
+			{
+				continue;
+			}
+
+			IECore::InternedString parameterName( &destName[splineElementMatch.position(1)], splineElementMatch.length(1) );
+			auto findParameter = origParameters.find( parameterName );
+			if( findParameter == origParameters.end() )
+			{
+				continue;
+			}
+
+			const SplinefColor3fData* colorSplineData = runTimeCast<const SplinefColor3fData>( findParameter->second.get() );
+			const SplineffData* floatSplineData = runTimeCast<const SplineffData>( findParameter->second.get() );
+
+			if( !( colorSplineData || floatSplineData ) )
+			{
+				continue;
+			}
+
+			InternedString adapterHandle;
+			int knotOffset;
+
+			// Insert a conversion shader to handle connection to component
+			auto inserted = currentSplineArrayAdapters.insert( { parameterName, std::make_pair( IECore::InternedString(), 0 ) } );
```

Comment:
This is another spot where structured bindings can add clarity (over things like `inserted.first->second.second`!).